src/client/rest/RESTMethods.js

```diff
@@ -522,15 +521,17 @@ class RESTMethods {
     return new Promise((resolve, reject) => {
       this.rest.makeRequest('post', `${Constants.Endpoints.guildEmojis(guild.id)}`, true, { name: name, image: image })
       .then(data => {
-        resolve(new Emoji(guild, data));
+        resolve(this.rest.client.actions.EmojiCreate.handle(data, guild).emoji);
       }).catch(reject);
     });
   }
 
-  deleteEmoji(guild, id) {
+  deleteEmoji(emoji) {
     return new Promise((resolve, reject) => {
-      this.rest.makeRequest('delete', `${Constants.Endpoints.guildEmojis(guild.id)}/${id}`, true)
-      .then(resolve).catch(reject);
+      this.rest.makeRequest('delete', `${Constants.Endpoints.guildEmojis(emoji.guild.id)}/${emoji.id}`, true)
+      .then(() => {
+        resolve(this.rest.client.actions.EmojiDelete.handle(emoji));
```

Comment:
should resolve `this.rest.client.actions.EmojiDelete.handle(emoji).data` by the looks of it, also make data less vague in the action